nwb_conversion_tools/utils/nwbephyswriterdatachunkiterator.py

```diff
@@ -18,12 +30,50 @@ def __init__(
         chunk_mb: float = None,
         chunk_shape: tuple = None,
     ):
+        """
+        Initialize an Iterable object which returns DataChunks with data and their selections on each iteration.
+
+        Parameters
+        ----------
+        ephys_writer : si.BaseRecording
+          The BaseRecording object (from spikeinterface/spikeinterface) which handles the API for data transfer.
+        segment_index : int, optional
+          DESCRIPTION. The default is 0.
+        unsigned_coercion : list, optional
+          For unsigned data types (e.g., uint16), this value shifts the dtype to a signed type.
+          The default is None.
```

Comment:
@alejoe91 Does the new spikeinterface object not have an offset field? If so, why are we asking for it to be set explicitly?

Reply:
It does. I think that we should set `unsigned_coercion` only if the dtype of the raw data is unsigned. What do you think?

Reply:
I would agree, and we should always check the offset field before that. Basically, the offset can serve two purposes

1) shifting the values of the series to some scientific unit
2) shifting the data type from unsigned to signed

since we don't know which is the use case ahead of time, but we do know the object has internally set offsets, we can start by using those. In pseudo-code

```
shifted_traces=unsigned_traces + internally set offset
if dtype(shifted_traces) are still unsigned:
    signed_traces = shifted_traces - 2**(n_bits-1)  # forces to be signed
```

where `n_bits=16` for `int16`, etc. 

Note that all this will only be an issue until the offset field becomes built-in for the PyNWB containers; once it is we can just set it there and write the series as is.

Reply:
Another problem is that when we apply the `unsigned_coercion` we are actually removing the offset, but the offset property is saved when adding electrodes...Also currently, the `NwbEphysWriterDataChunkIterator` returns an unsigned dtype even when an `unsigned_coercion` is used. In that case, the `_get_dtype` function should return a signed dtype. Right?

Reply:
> offset property is saved when adding electrodes
We can probably stop doing that until we add it as a specific attribute assigned to datasets (long-term plan); it's enough now to simply take an unsigned type and make it a signed type.

> Also currently, the NwbEphysWriterDataChunkIterator returns an unsigned dtype even when an unsigned_coercion is used. In that case, the _get_dtype function should return a signed dtype. Right?
The way I'd see it here is in that case, we have the `dtype` that goes into the NWB conversion process (say, `uint16`), but the `dtype` we want to save it as in the NWBFile would be the signed version (say, `int16`). The only added cost is to doing the addition/subtraction at each buffer read, but from personal experiments on integer data types that isn't too costly an operation.

Reply:
> Another problem is that when we apply the unsigned_coercion we are actually removing the offset,

The way I have it outlined it above, the information as an attribute goes away but the information is instead absorbed into the series itself WLOG; even the `itemsize` is actually preserved except in an edge case where there was (a) the maximum value of the data itself was on the right edge of the unsigned range and (b) the offset was a non-zero integer.

Or I guess if the offset itself is a float is kind of the worst case scenario, but I've never seen that before. You can maybe add a prohibition of that until we can set the `offsett` as an NWB attribute.

Reply:
I'm working on a solution! :) Seems to work locally but it needs some more testing. I'll push soon